gridded_etl_tools/utils/attributes.py

```diff
@@ -83,217 +84,148 @@ def host_organization(self) -> str:
+    @deprecation.deprecated("Use the missing_value attribute")
     def missing_value_indicator(cls) -> str:
-        """
-        Default indicator of a missing value in a dataset
-
-        Returns
-        -------
-        str
-           Stand-in for a missing value
+        return cls.missing_value
 
-        """
-        return ""
-
-    @property
-    def tags(cls) -> list[str]:
-        """
-        Default tag for a dataset. Prevents crashes on parse if no tags assigned.
+    tags: list[str] == [""]
+    """
+    Tags for dataset.
+    """
 
-        Returns
-        -------
-        list[str]
-           Stand-in for a dataset's tags
+    forecast: bool = False
+    """
+    ``True`` if the dataset provides forecast data.
+    """
 
-        """
-        return [""]
-
-    @property
-    def forecast(self) -> bool:
-        """Forecast defaults to False, must override for actual forecast datasets"""
-        return False
-
-    @property
-    def ensemble(self) -> bool:
-        """Ensemble defaults to False, must override for actual ensemble datasets"""
-        return False
-
-    @property
-    def hindcast(self) -> bool:
-        """Hindcast defaults to False, must override for actual hindcast datasets"""
-        return False
-
-    @property
-    def forecast_hours(self) -> list[int]:
-        """To be overwritten by actual forecast datasets"""
-        return list(None)
-
-    @property
-    def ensemble_numbers(self) -> list[int]:
-        """To be overwritten by actual ensemble datasets"""
-        return list(None)
-
-    @property
-    def hindcast_steps(self) -> list[int]:
-        """To be overwritten by actual hindcast datasets"""
-        return list(None)
+    ensemble: bool = False
+    """
+    ``True`` if the dataset provides ensemble data.
+    """
 
-    @classmethod
-    def irregular_update_cadence(self) -> None | tuple[np.timedelta64, np.timedelta64]:
-        """
-        If a dataset doesn't update on a monotonic schedule return a tuple noting the lower and upper bounds of
-        acceptable updates Intended to prevent time contiguity checks from short-circuiting valid updates for datasets
-        with non-monotic update schedules
-        """
-        return None
+    hindcast: bool = False
+    """
+    ``True`` if the dataset privides hindcast data.
+    """
 
-    @property
-    def bbox_rounding_value(self) -> int:
-        """
-        Value to round bbox values by. Specify within the dataset for very high resolution datasets
-        to prevent mismatches with rounding behavior of old Arbol API.
+    forecast_hours: list[int] = []
+    """"
+    Hours provided by the forecast, if any.
+    """
 
-        Returns
-        -------
-        int
-            The number of decimal places to round bounding box values to.
-        """
-        return 5
+    ensemble_numbers: list[int] = []
+    """
+    Numbers uses for ensemble, if any.
+    """
 
-    @property
-    def store(self) -> StoreInterface:
```

Comment:
Note that I removed the getter/setter for store because that is already handled well in the constructor for DatastoreManager.

Reply:
That's true but at least with the current design or lack of design the store can be changed during the lifetime of the object

Reply:
Is that a common use case?

Reply:
It's not and maybe even shouldn't be but in general the library runs in the interpreter and notebooks more often than originally, so we've been tightening up the member properties like keeping the store from being set to something arbitrary

Reply:
e99bba7

Reply:
:pray: